---
title: "EMA_Microarray_Analysis"
author: "Tzu L. Phang"
date: "April 9, 2015"
output:
  pdf_document:
    latex_engine: lualatex
---
# Idiopathic Pulmonary Arterial Hypertension Gene Expression Analysis

## Background

- Pulmonary Hypertension (PH) is a rare disease marked by abnormally high blood pressure in the arteries of the lungs. Pulmonary Arterial Hypertension (PAH) is a form of PH which causes increased stress to the heart and causes a rise in blood pressure within the lungs [1-2]. PAH is more common in females than males [3] and symptoms of PAH present earlier in life (30-40 years of age) rather than later (40-50 years of age).
- There is no known cure for PAH at this time. PAH is most commonly treated with medications designed to reduce the work of the heart. In addition to medication, PAH is treated surgically with an atrial septostomy or through a lung transplant [3]. If left untreated, PAH results in serious illness or death, thus it is crucial to gain a better understanding of the disease [2].
- Of the four different types of PAH patients, this analysis utilizes samples collected from patients with Idiopathic Pulmonary Arterial Hypertension. This type of PAH occurs in individuals with no family history of the disease; there is no known genetic mutation [4].
- The purpose of the study is to gain genetic insight to better understand potential genetic causes of disease.
- The goal of the current analysis is to identify and examine the gene expression signature for IPAH patients compared to the controls (Failed Donors; FD).

### References:
1. Pulmonary hypertension. American Heart Association. http://www.heart.org/HEARTORG/Conditions/CongenitalHeartDefects/TheImpactofCongenitalHeartDefects/Pulmonary-Hypertension_UCM_307044_Article.jsp
2. Pulmonary arterial hypertension. American Lung Association. http://www.lung.org/lung-disease/pulmonary-arterial-hypertension/
3. Rajkumar, R., Konishi, K., Richards, T. J., Ishizawar, D. C., Wiechert, A. C., Kaminski, N., & Ahmad, F. (2010). Genomewide RNA expression profiling in lung identifies distinct signatures in idiopathic pulmonary arterial hypertension and secondary pulmonary hypertension. American Journal of Physiology-Heart and Circulatory Physiology, 298(4), H1235.
4. Pulmonary arterial hypertension. Genetics Home Reference. http://ghr.nlm.nih.gov/condition=pulmonaryarterialhypertension



## Experimental Design

Made some changes on April 29, 2015, push again .... !!?

Read in descriptive data file and run summary statistics to examine the distribution of the data prior to importing CEL files

```{r, cache=TRUE, results='asis', message=FALSE}
options(width=500)
## Load EMA and oligo libraries
library(EMA)
library(oligo)

## Read in descriptive data file
PBHI.file.info = read.table(file = "./data/PHBI cel annotation BIO6660 v2.txt", header = T, sep = "\t")
## print(PBHI.file.info)
knitr::kable(PBHI.file.info)
```

We will use the "Clinical.Category" column to define the 2 comparing groups

```{r, cache=TRUE, results='markup', message=FALSE}
## Extract groups: IPAH and Failed Donor
PBHI.type.cl = as.character(PBHI.file.info$Clinical.Category)
PBHI.type.cl
```

We suspect there might be batch-effect.  We will use the "Batch" column to extract potential batch-effect information

```{r, cache=TRUE, results='markup', message=FALSE, warning=FALSE}
PBHI.batch.cl = PBHI.file.info$Batch
PBHI.batch.cl
```

What is the average age for the 2 groups of patients

```{r, cache=TRUE, results='markup', message=FALSE}
## Run basic summary statistics
tapply(PBHI.file.info$Age, PBHI.file.info$Clinical.Category, mean)
```

What is the gender distribution

```{r, cache=TRUE, results='markup', message=FALSE}
tapply(PBHI.file.info$Sex, PBHI.file.info$Clinical.Category, summary)
```

### Visualize Age Distribution by Group

```{r, cache=TRUE, results='markup', message=FALSE}
library(ggplot2)

plot = ggplot(PBHI.file.info, aes(factor(PBHI.file.info$Clinical.Category), 
    PBHI.file.info$Age)) + geom_boxplot(aes(fill = factor(PBHI.file.info$Clinical.Category))) + 
    labs(title = "Age Distributions by PAH Group") + xlab("Group") + ylab("Age")
plot + scale_fill_manual("PAH Group", values = c("#56B4E9", "#009E73"))
```

## Preprocessing and Quality Assessemnt

### Import CEL files

Import CEL files, read in data, and normalize data using the RMA function

```{r, cache=TRUE, results='markup', message=FALSE, warning=FALSE, }
## Import CEL files
PBHI.CEL = PBHI.file.info$PBHI.CEL
PBHI.CEL = list.celfiles('./data')
PBHI.CEL = paste('./data/', PBHI.CEL, sep = '')

## Read CEL files to directory
PHBI.data = read.celfiles(PBHI.CEL, verbose = F)

## Normalize the data
PHBI.norm = rma(PHBI.data)
```


### Visual Inspection: Before and After Data Normalization

Create boxplots of log-intensity distribution to visualize data pre and post normalization

```{r, cache=TRUE, results='markup', message=FALSE, warning=FALSE}
## Load color libraries
library(RColorBrewer)

## Set color palette
color.palette = brewer.pal(8, "Set1")


## Pre-normalized intensity values boxplot
boxplot(PHBI.data, col = color.palette, main = "Pre-normalized Intensity Values")

## Normalized intensity values boxplot
boxplot(PHBI.norm, col = color.palette, main = "Normalized Intensity Values")


```


Create density plots of log-intensity distribution to visualize data pre and post normalizatio

```{r, cache=TRUE, results='markup', message=FALSE, warning=FALSE}
## Pre-normalized density plot of log-intensity distribution
hist(PHBI.data, col = color.palette, main = "Pre-Normalized Density Plot of log-Intensity Distribution")

## Normalized density plot of log-intensity distribution
hist(PHBI.norm, col = color.palette, main = "Normalized Density Plot of log-Intensity Distribution")

```

### Filtering Process

Extract normalized expression values and perform filtering to discard probesets below a specified threshold

```{r, cache=TRUE, results='markup', message=FALSE, warning=FALSE}
## Extract expression values after normalization, verify dimensions
PHBI.exprs = exprs(PHBI.norm)
dim(PHBI.exprs)

## Filter and discard probesets with a maximum log2 expression value below 4,
## p=0.01
PUBHI.f = expFilter(PHBI.exprs, threshold = 4)

## View data ditribution after filtering; remove threshold line
PUBHI.A = expFilter(PUBHI.f, threshold = F)
```

### Hierarchical Clustering

```{r, cache=TRUE, results='markup', message=FALSE, warning=FALSE}
PUBHI.sample = clustering(data = PUBHI.f, metric = "pearson", method = "ward")

## Visualize data pre and post filtering
clustering.plot(tree = PUBHI.sample, lab = PBHI.type.cl, title = "Filtered Data")

clustering.plot(tree = PUBHI.sample, lab = PBHI.batch.cl, title = "Filtered PBHI Data by Batch")
```

### Principal Component Analysis

Run Principle Component Analysis (PCA) on the normalized and filtered data

```{r, cache=TRUE, results='markup', message=FALSE, warning=FALSE}
## Segment variation into different components
acp = runPCA(t(PUBHI.f), scale = FALSE, lab.sample = PBHI.type.cl, plotSample = FALSE, 
    plotInertia = FALSE)
plotInertia(acp)

## View pre-batch correction individual maps (axes 1 and 2) to look at the
## variation between groups and batches
plotSample(acp, axes = c(1, 2), lab = PBHI.type.cl)

plotSample(acp, axes = c(1, 2), lab = as.character(PBHI.batch.cl))

## Create pdf report of PCA with selected plots
acp = runPCA(t(PUBHI.f), scale = FALSE, pdfname = "PCA.pdf", lab.sample = PBHI.type.cl)

```

### Batch Correction

Perform batch correction on the normalized data and re-run PCA to show post-batch correction improvements

```{r, cache=TRUE, results='markup', message=FALSE, warning=FALSE}
# Load sva library
library(sva)

## Create experimental design object and perform batch correction
mod = model.matrix(~as.factor(PBHI.type.cl))
combat_edata = ComBat(dat = PUBHI.f, batch = PBHI.batch.cl, mod = mod, numCovs = NULL, 
    par.prior = TRUE, prior.plots = FALSE)
```

Run PCA and Hierarchical Clustering again after batch adjustment

```{r, cache=TRUE, results='markup', message=FALSE, warning=FALSE}
## Run PCA after batch correction
acp = runPCA(t(combat_edata), scale = FALSE, lab.sample = PBHI.type.cl, plotSample = FALSE, plotInertia = FALSE)
plotInertia(acp)

## View post-batch correction individual maps (axes 1 and 2) to look at the
## variation between groups and batches
plotSample(acp, axes = c(1, 2), lab = PBHI.type.cl)

plotSample(acp, axes = c(1, 2), lab = PBHI.batch.cl)

## Sample Hierarchical Clustering (post-batch adjustment)
PUBHI.sample2 = clustering(data = combat_edata, metric = "pearson", method = "ward")
clustering.plot(tree = PUBHI.sample2, lab = PBHI.type.cl, title = "PHBI Data - Filtered")
clustering.plot(tree = PUBHI.sample2, lab = PBHI.batch.cl, title = "PHBI Data By Batch - Filtered")
```

## Statistical Analysis

Run Student's t-test and Significance of Analysis of Microarrays (SAM) test on normalized data. Merge the test results into one data set to produce a heatmap

```{r, cache=TRUE, results='markup', message=FALSE, warning=FALSE}
## Set-up Student's t-test comparison factor
PUBHI.type.num = ifelse(PBHI.type.cl == "IPAH", 0, 1)

## Run Student's t-test with batch corrected data
PUBHI.ttest = runTtest(combat_edata, labels = PUBHI.type.num, algo = "t.equalvar", 
    q = 0.2, plot = FALSE)

## head(PUBHI.ttest)
knitr::kable(head(PUBHI.ttest))

## Run the SAM test with batch corrected data
PUBHI.SAM = runSAM(PUBHI.f, labels = PUBHI.type.num)

knitr::kable(head(PUBHI.SAM))

```
Reorganize the result 

```{r, cache=TRUE, results='markup', message=FALSE, warning=FALSE}
## Sort genes in both datasets by the 'probeID' variable
PUBHI.ttest.sig = PUBHI.ttest[order(PUBHI.ttest$probeID), ]
PUBHI.SAM.sig = PUBHI.SAM[order(PUBHI.SAM$probeID), ]

## Extract the 'probeID' and 'FoldChange' variables from the PBHI.SAM.sig
## data set
FC.raw = PUBHI.SAM.sig[, c("probeID", "FoldChange")]

## Merge the 'FoldChange' Variable into PUBHI.ttest.sig database by probeID
PUBHI.test.merge = merge(PUBHI.ttest.sig, FC.raw, by = "probeID", all.y = TRUE, 
    all.x = TRUE)
knitr::kable(head(PUBHI.test.merge))

## Re-sort merged data by the 'AdjpValue' variable
PUBHI.test.merge = PUBHI.test.merge[order(PUBHI.test.merge$AdjpValue), ]
knitr::kable(head(PUBHI.test.merge))

```

### Volcano Plot

```{r, cache=TRUE, results='markup', message=FALSE, warning=FALSE}
## Produce a volcano Plot to verify SAM findings
volcano = ggplot(data = PUBHI.test.merge, aes(x = log2(PUBHI.test.merge$FoldChange), 
    y = -log10(PUBHI.test.merge$AdjpValue)), colour = none) + geom_point(alpha = 0.4, 
    size = 1.75) + labs(title = "PAH Volcano Plot") + xlim(c(-2, 2)) + ylim(c(0, 
    2)) + xlab("Log2 Fold Change") + ylab("-Log10 P-Values")
volcano
```

## Output Analysis Result

Produce a heatmap of the top 100 significant genes

```{r, cache=TRUE, results='markup', message=FALSE, warning=FALSE}
## Produce Heatmap
mvgenes = as.character(PUBHI.test.merge$probeID[1:100])
c.sample <- clustering(data = PUBHI.f[mvgenes, ], metric = "pearson", method = "ward")
c.gene <- clustering(data = t(PUBHI.f[mvgenes, ]), metric = "pearson", method = "ward")
clustering.plot(tree = c.sample, tree.sup = c.gene, data = PUBHI.f[mvgenes, 
    ], names.sup = FALSE, lab = PBHI.type.cl, trim.heatmap = 0.99)

```


### Annotate Significant Genes

Annotate the top 250 significant genes, output to a text file. Input cluster analysis output from DAVID and print the first three rows

```{r, cache=TRUE, results='markup', message=FALSE, warning=FALSE}
options(width = 500)
## Load annaffy and hugene10sttranscriptcluster.db libraries
library(annaffy)
library(hugene10sttranscriptcluster.db)

## Annotate the top 250 significant genes
anntable = aafTableAnn(as.character(PUBHI.test.merge$probeID[1:250]), "hugene10sttranscriptcluster.db")

## Add the 'AdjpValue' and 'FoldChange' variables to the annotation table
atable = aafTable(`P-Value` = PUBHI.test.merge$AdjpValue[1:250], signed = TRUE)
FCtable = aafTable(`Fold Change` = PUBHI.test.merge$FoldChange[1:250], signed = TRUE)
table = merge(anntable, atable)
table2 = merge(table, FCtable)

## Export results to an HTML and text file
saveHTML(table2, file = "PBHI.psig.genes.htm")
saveText(table2, file = "PBHI.psig.genes.txt")

## Print top 7 significant genes at p ≤ 0.01 level
#annot.output = read.table(file = "annot.sig.txt", header = T, sep = "\t", nrows = 7)
#annot.outputs = annot.output[order(annot.output$P.Value), ]
#knitr::kable(annot.outputs)
```

R session info

```{r, cache=TRUE, results='markup', message=FALSE, warning=FALSE}
sessionInfo()
```












